### PR TITLE
Add module for micrometer/prometheus

### PR DESCRIPTION
**Pull request by Sgitario**
Verifies that custom metrics are exposed in the embedded Prometheus instance provided by OpenShift.

There is a PrimeNumberResource that checks whether an integer is prime or not. The application will expose the following custom metrics:
- `prime_number_max_{uniqueId}`: max prime number that was found
- `prime_number_test_{uniqueId}`: with information about the calculation of the prime number (count, max, sum)

Where `{uniqueId}` is an unique identifier that is calculated at startup time to uniquely identify the metrics of the application.

### CODE REVIEW

**Review by mjurc — APPROVED** [review]
LGTM!

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

File: micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private final LongAccumulator highestPrime = new LongAccumulator(Long::max, 0);
+    private final MeterRegistry registry;
+    private final String uniqueId = RandomStringUtils.randomAlphabetic(5);
+
+    PrimeNumberResource(MeterRegistry registry) {
+        this.registry = registry;
+
+        // Create a gauge that uses the highestPrimeNumberSoFar method
+        // to obtain the highest observed prime number
+        registry.gauge("prime.number.max." + uniqueId, this,
```
Comment (pjgg) [inline micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java:27]: Not sure if its a good idea to create a metric with a Unique Key.  I will suggest use as a key "prime.number.max" and then add tags . So, in this way, we could make queries over "prime.number.max"  and then filter by tag.

File: micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            for (int i = 3; i < Math.floor(Math.sqrt(number)) + 1; i = i + 2) {
+                if (number % i == 0) {
+                    return number + " is not prime, is divisible by " + i + ".";
+                }
+            }
+            highestPrime.accumulate(number);
+            return number + " is prime.";
+        };
+
+        return registry.timer("prime.number.test." + uniqueId).wrap(supplier).get();
```
Comment (pjgg) [inline micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java:65]: Add uniqueId as a tag. (sugestion)

Example:

`registry.timer("prime.number" , "DEV", uniqueId)`
`registry.timer("prime.number" , "TEST", uniqueId)`

**Review by pjgg — CHANGES_REQUESTED** [review]

File: micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            return "1 is not prime.";
+        }
+        if (number == 2) {
+            return "2 is prime.";
+        }
+        if (number % 2 == 0) {
+            return number + " is not prime, it is divisible by 2.";
+        }
+
+        Supplier<String> supplier = () -> {
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java:55]: I wanted to replicate exactly what was done in here: https://quarkus.io/guides/micrometer
This way we can spot issues in the guide over time.

File: micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java (hunk 1 above)
Comment (Sgitario) [reply micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java:27]: Metrics will be deleted after some time from Prometheus. 

Using a unique id will extremely ease the test verification (see how we assert the metrics).

File: micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java (hunk 2 above)
Comment (Sgitario) [reply micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java:65]: Using a unique id extremely ease the test verification (see how we assert the metrics).

**Review by pjgg — APPROVED** [review]

**Pull request merged**